src/cockpit/packages.py

```diff
@@ -159,12 +169,25 @@ def sortify_version(version: str) -> str:
         return '.'.join(part.zfill(8) for part in version.split('.'))
 
     def add_file(self, item: Path, checksums: List['hashlib._Hash']) -> None:
-        rel = item.relative_to(self.path)
-        self.files.add(rel)
+        rel = str(item.relative_to(self.path))
+        guessed_type = mimetypes.guess_type(rel)
 
         with item.open('rb') as file:
             data = file.read()
 
+        # Keep the file in memory to serve it later: if this is a 'po.*.js'
```

Comment:
FTR, I prefer re-reading it from disk. If there's enough RAM, the kernel would already cache that anyway, and with memory pressure it has more chances to drop caches. Of course building the `self.translations` map is a good idea, but it could perhaps just map to the computed file name, instead of the file contents?

Not a hard "no", just a tought. You mentioned "14 MB", which isn't small, but these days it's not unreasonable either.

Reply:
Some thoughts:

 - the main issue here is that I want the file serving code to always do the same thing.  "Open this filename and serve the bytes" doesn't work as that thing, since we need to serve an empty file for English po.js (unless we want to use `/dev/null` as a filename)
 - my ideal approach would be to `mmap()` this, which would let the kernel drop it if it needed to, but allow us to serve it directly, but this isn't the safest option: if someone `O_TRUNC`s the file, we get `SIGBUS`, and I don't really feel like trying to sort that out in Python
 - we could also keep the fd around, but see above notes about `/dev/null`

Reply:
.. or we could just actually create a `po.en.js` to get rid of that special case.

Either way, feel free to resolve -- this feels like it's well into the "optimize later if ever necessary" category, especially since going from the C to Python bridge increased memory usage by a *lot* anyway. :grin: